backend/src/core/services/google-auth/google-create-or-login-user.usecase.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@nestjs/common';
+import { UserMapper } from '../user/user.mapper';
+import { UserDao } from '@src/data/dao/user/user.dao';
+import { GoogleUserRequest } from '@shared/request-response/user/google-create-user.request';
+import { UserEntity } from '../user/entities/user.entity';
+import { BaseEntity } from '@src/core/abstraction/base-entity';
+import { RefreshSessionService } from '../auth/services/refresh-session.service';
+import { AccessTokenService } from '../auth/services/access-token.service';
+import { REFRESH_TOKEN_LIFETIME_IN_SECONDS } from '../auth/constants';
+import { NotFoundException } from '@src/core/exceptions/not-found.exception';
+import { RefreshSessionEntity } from '../auth/entities/refresh-session.entity';
+import { LoginSuccessDto } from '../auth/dto/login-success.dto';
+
+@Injectable()
+export class GoogleCreateOrLoginUserUsecase {
+  constructor(
+    private readonly userDao: UserDao,
+    private readonly userMapper: UserMapper,
+    private readonly refreshSessionService: RefreshSessionService,
+    private readonly accessTokenService: AccessTokenService,
+  ) {}
+  /* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async execute(request: any, googleUserRequest: GoogleUserRequest): Promise<LoginSuccessDto> {
```

Comment:
Do not pass request object to the core usecese. It violates layered architecture (core layer should not know about http). Create and use LogigGoogleDto